gui-apps/waybar/waybar-0.7.1.ebuild

```diff
@@ -0,0 +1,60 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit meson
+
+DESCRIPTION="Highly customizable Wayland bar for Sway and Wlroots based compositors."
+HOMEPAGE="https://github.com/Alexays/Waybar"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/Alexays/${PN^}.git"
+else
+	SRC_URI="https://github.com/Alexays/${PN^}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="libcxx mpd network pulseaudio tray udev"
+
+DEPEND="
+	dev-cpp/gtkmm:3.0
+	dev-libs/jsoncpp
+	dev-libs/libinput
+	dev-libs/libsigc++:2
+	>=dev-libs/libfmt-5.3.0
+	>=dev-libs/spdlog-1.3.1
+	dev-libs/wayland
+	dev-libs/wayland-protocols
+	gui-libs/wlroots
+
+	libcxx? ( sys-libs/libcxx )
+	mpd? ( media-libs/libmpdclient )
+	network? ( dev-libs/libnl:3 )
+	pulseaudio? ( media-sound/pulseaudio )
+	tray? ( dev-libs/libdbusmenu[gtk3] )
+	udev? ( virtual/libudev )
+"
+RDEPEND="
+	${DEPEND}
```

Comment:
You should put `:=` slot rules on dependencies it's linking with and that have soname versionning. For example jsoncpp with `jsoncpp:0/19` providing `libjsoncpp.so.19`.

Reply:
How do I figure out the correct soname version? I tried listing all the linked libraries on my exec, but I'm not sure I should use it as a reference: https://gist.github.com/lovesegfault/88e23dccb2aa0bdea0d3cccf664d0318

Reply:
IMHO just putting a `:=` on the existing deps that you have should be quite okay, specially if they end in `.0` and/or have no (sub)slot in gentoo yet.

Reply:
Done! :)

Reply:
subslots trigger rebuilds should the soname change from the package providing it.  That is the most important factor.